sklearn/utils/_testing.py

```diff
@@ -788,6 +797,10 @@ def assert_docstring_consistency(
         parameters/attributes/returns. If None, will revert to default behavior
         of comparing descriptions between objects.
 
+    ignore_types : list, default=[]
+        List of parameter/attribute/return names to exclude from matching type
+        between objects.
```

Comment:
Is this what it does? I think it would be helpful to have the example in the docstring here to be clear

```suggestion
        List of parameter/attribute/return names to exclude from matching type
        between objects. For instance, the following two would be accepted:
        
        .. code
          param : str, None
            param description
            
          param : str
            param description
            
```

Reply:
Yes that's exactly what it would do.

Do you prefer this to https://github.com/scikit-learn/scikit-learn/pull/30943 ? It is much simpler and I think I would prefer it for this reason.

Reply:
Yes I do.